### PR TITLE
Ensure deletes are applied in the right order for Synapse Link

### PR DESCRIPTION
**Pull request by george-zubrienko**
## Scope
- Fix edge case when deletes can be applied before the insert if they fall into different batches
- Added unit test for deletes

### CODE REVIEW

**Review by s-vitaliy — APPROVED** [review]

File: src/main/scala/services/synapse/base/SynapseLinkReader.scala [hunk 1]
```diff
@@ -69,6 +67,9 @@ final class SynapseLinkReader(entityName: String, storagePath: AdlsStoragePath,
     )
   )
 
+  private def getEntityDeletes(startDate: OffsetDateTime): ZStream[Any, Throwable, SchemaEnrichedBlob] = getEntityChangeData(startDate).filter(seb => seb.blob.name.endsWith("/1.csv"))
```
Comment (s-vitaliy) [inline src/main/scala/services/synapse/base/SynapseLinkReader.scala:70]: Write a comment about why are we doing the split here.

**Review by s-vitaliy — APPROVED** [review]
